hibernate-core/src/main/java/org/hibernate/query/sqm/internal/ConcreteSqmSelectQueryPlan.java

```diff
@@ -188,6 +188,20 @@ public ConcreteSqmSelectQueryPlan(
 		//		`#performList` and `#performScroll`.
 	}
 
+	private static void autoFlushIfRequired(
+			JdbcOperationQuerySelect jdbcSelect,
+			SharedSessionContractImplementor session) {
+		if ( jdbcSelect.dependsOnParameterBindings() ) {
```

Comment:
I don't understand this condition. Shouldn't this rather use the same condition as in `buildCacheableSqmInterpretation` to decide whether to do a pre-flush?